src/X.Serilog.Sinks.Telegram/Batch/Rules/OncePerTimeRule.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading;
+
+namespace X.Serilog.Sinks.Telegram.Batch.Rules;
+
+public class OncePerTimeRule : IRule, IExecutionHook
+{
+    private readonly TimeSpan _delay;
+
+    private DateTime _nextExecution;
+
+    public OncePerTimeRule(TimeSpan delay)
+    {
+        _delay = delay;
+        _nextExecution = Now + delay;
+    }
+
+    private static DateTime Now => DateTime.Now;
```

Comment:
That's wieerd helper property.

There are no static import, like in Java in C#? :)

Reply:
But OI see nothing bad in calling DateTime.Now in code.

But on other hand You might consider using IClock instead. As actual field. So clock can be provided inside tests.

Sadly looks like it doesn't exist in stdlib, but there are packages https://github.com/RobThree/IClock

Reply:
Sonething from MS https://learn.microsoft.com/en-us/dotnet/api/orleans.transactions.iclock?view=orleans-7.0